test/Table_Tests/src/Common_Table_Operations/Column_Operations_Spec.enso

```diff
@@ -1165,11 +1165,20 @@ add_column_operation_specs suite_builder setup =
 
         if setup.is_database.not then
             group_builder.specify "ceil, floor and truncate should work correctly on Floats outside the java Long range" <|
-                positive_values = [9223372036854775000.0, 9223372036854776000.0, 9223372036854775807000000.0]
+                positive_values = [9223372036854774784.0, 9223372036854776000.0, 9223372036854775807000000.0]
                 values = positive_values + positive_values.map .negate
-                values.map x->
+                values.map x-> Test.with_clue "(x="+x.to_text+") " <|
                     x_int = x.truncate
+
+                    # Test assumptions
+                    x.should_be_a Float
+                    x_int.should_be_a Integer
+                    # Relies on float-integer equality: 2 == 2.0
+                    x_int.should_equal x
+
                     c = table_builder [["x", [x, -x]]] . at "x"
+                    c.value_type.should_equal Value_Type.Float
+
                     c.ceil . to_vector . should_equal [x_int, -x_int]
```

Comment:
This test started failing because before the column `c` was inferred to be a column of only integers, and thus most likely it had type `Decimal` (not `Integer` because some values did not fit in `int64`). Thus the `truncate` etc. operations were just a no-op.

Now the column is constructed as `Float` and this test finally does something. And it uncovered our implementations of `truncate` (and `floor`/`ceil`) are inconsistent between the builtin base method and the Table column operation.

Reply:
The number `9223372036854775000.0` would never pass the test under normal circumstances because as `double` it is actually represented as `9223372036854774784`, so this is the value we get once we truncate and obviously it is _not_ equal to `9223372036854775000.0`.

The test used to pass only because immediately on column construction the `% 1.0 == 0.0` float value was converted to integer, so it already was equal to `9223372036854774784` before the `Column.truncate` operation was even called - so the test succeeded trivially.

Now that the test actually does something this misalignment showed up, so instead I adapted this value to match the behaviour. Dealing with floating point imprecision at large magnitudes is not the point of this test.